jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/WorkflowProcessImpl.java

```diff
@@ -116,4 +116,24 @@ protected void processNodeContainer(org.jbpm.workflow.core.NodeContainer nodeCon
             }
         }
     }
+
+    protected Node getContainerNode(Node currentNode, org.jbpm.workflow.core.NodeContainer nodeContainer, long nodeId) {
+        for (Node node : nodeContainer.getNodes()) {
+            if (nodeId == node.getId()) {
+                if (currentNode != null) {
+                    return currentNode;
```

Comment:
Maybe just one thing when I looked at this the second time. Is this check really needed? If the `currentNode` is `null`, then we can immediately return `null` because given that nodeId is unique, it won't match any of these conditions next time, so it will eventually reach `return null;`.